kong/llm/proxy/handler.lua

```diff
@@ -344,12 +346,15 @@ function _M:header_filter(conf)
   if llm_state.is_streaming_mode() then
     -- we are going to send plaintext event-stream frames for ALL models
     kong.response.set_header("Content-Type", "text/event-stream")
-  end
-
-  if accept_gzip() then
-    kong.response.set_header("Content-Encoding", "gzip")
+    -- TODO: disable gzip for SSE because it needs immediate flush for each chunk
+    -- and seems nginx doesn't support it
   else
-    kong.response.clear_header("Content-Encoding")
+
+    if accept_gzip() then
```

Comment:
or use elseif here?

Reply:
That would be slightly not equivalent to what's happening here. The last "else" would always happen. Instead it seems we only want `kong.response.clear_header("Content-Encoding")` to happen if `llm_state.is_streaming_mode()` is false (and not when it is non-falsy but accept_gzip is true)